src/containers/templates/Positive/index.js

```diff
@@ -48,59 +50,50 @@ const BubbleInner = styled.div`
   line-height: 1.5em;
 `;
 
+const random = (bottom, top) => Math.floor(Math.random() * top) + bottom;
+
 export default class extends React.Component {
   constructor(props) {
     super(props);
     this.state = { task: {} };
   }
 
-  componentDidMount() {
+  async componentDidMount() {
     this.userId = firebase.auth().currentUser.uid;
-
-    const random = (bottom, top) => Math.floor(Math.random() * top) + bottom;
-
     const questionsPath = 'aspirations/positive/turtle-questions';
 
-    this.getOrCreateLatestEntry().then(latestEntryId => {
-      this.entryId = latestEntryId;
-      const firebasePath = `/users/${this
-        .userId}/tasks/positive/entries/${latestEntryId}`;
-
-      this.binding = base.syncState(firebasePath, {
-        context: this,
-        state: 'task',
-        then: () => {
-          if (!this.state.task.question) {
-            base
-              .fetch(questionsPath + '/count', {
-                context: this
-              })
-              .then(count => {
-                this.questionNumber = random(0, count);
-
-                return base.fetch(questionsPath + '/' + this.questionNumber, {
-                  context: this
-                });
-              })
-              .then(question => {
-                this.setState(({ task }) => ({ task: { ...task, question } }));
-              })
-              .catch(error => {
-                console.error(error);
-              });
+    const latestEntryId = await this.getOrCreateLatestEntry();
+    this.entryId = latestEntryId;
+    const firebasePath = `/users/${this.userId}/tasks/positive/entries/${latestEntryId}`;
+
+    this.binding = base.syncState(firebasePath, {
+      context: this,
+      state: 'task',
+      then: async () => {
+        if (!this.state.task.question) {
+          try {
+            const count = await base.fetch(questionsPath + '/count', {
```

Comment:
@AlexGilleran , if you haven't known this `async` `await` syntax, it is pretty handy to avoid nested `then`